lib/spack/llnl/util/lock.py

```diff
@@ -195,8 +195,8 @@ def _lock(self, op, timeout=None):
             total_wait_time = time.time() - start_time
             return total_wait_time, num_attempts
 
-        raise LockTimeoutError("Timed out waiting for a {0} lock."
-                               .format(lock_type[op]))
+        raise LockTimeoutError("Timed out waiting for a {0} lock after {1}s."
```

Comment:
If we enter this branch of code, would the timeout be shown in the line above the error, per the log at line 177?

Reply:
Yes, but only if you run under `-d`, this is (at least for the cases where it annoyed me) a fatal error. I'm not sure when this occurred, but I typically don't use `spack -d` due to the flood of messages...

I don't have strong feelings, if you prefer to keep it as is, feel free to close the PR

Reply:
Closing the PR, but feel free to re-raise if you have a second thought on it.